src/cpp/proxy/sockjsmanager.cpp

```diff
@@ -241,9 +241,9 @@ class SockJsManager::Private : public QObject
 
 		s->route = route;
 
-		readyReadConnection = req->readyRead.connect(boost::bind(&Private::req_readyRead, this));
-		bytesWrittenConnection = req->bytesWritten.connect(boost::bind(&Private::req_bytesWritten, this, boost::placeholders::_1));
-		errorConnection = req->error.connect(boost::bind(&Private::req_error, this));
+		readyReadConnection = req->readyRead.connect(boost::bind(&Private::req_readyRead, this, req));
+		bytesWrittenConnection = req->bytesWritten.connect(boost::bind(&Private::req_bytesWritten, this, boost::placeholders::_1, req));
+		errorConnection = req->error.connect(boost::bind(&Private::req_error, this, req));
```

Comment:
There can be multiple requests at once.

Reply:
got it, thanks. will update to a connection map.
to be sure I am not missing another other connections, could we have the same issue here:
https://github.com/fastly/pushpin/blob/0cca88e269056bd930f89c32cbf253e0a82724b6/src/cpp/proxy/updater.cpp#L122
or here: 
https://github.com/fastly/pushpin/blob/0cca88e269056bd930f89c32cbf253e0a82724b6/src/cpp/proxy/websocketoverhttp.cpp#L614

Reply:
those are fine

Reply:
@jkarneges your thoughts are appreciated here

Reply:
https://github.com/fastly/pushpin/pull/47894#discussion_r1467980837 :)

Reply:
somehow I was not seeing that earlier message until now! thanks justin for your patience :)